ops/testing.py

```diff
@@ -1146,34 +1148,44 @@ def add_network(self, binding_name: str, address: str, *,
                     relation_id: Optional[int] = None,
                     cidr: Optional[str] = None,
                     interface: str = 'eth0',
-                    ingress_addresses: Optional[List[str]] = None,
-                    egress_subnets: Optional[List[str]] = None,
+                    ingress_addresses: Optional[Sequence[str]] = None,
```

Comment:
I think `Sequence` would not allow things like a generator or iterator here, is that intentional? https://docs.python.org/3/glossary.html#term-sequence

Iterators/ generators don't implement `__getitem__` nor `__len__` I believe

Reply:
The most generic type hint would be `Iterable`: https://docs.python.org/3/library/typing.html#typing.Iterable. This would also allow `list`, `tuple`, `set`, ...

Reply:
`typing.Sequence` and `typing.Iterable` are deprecated in favour of the `ABC` btw, although depending on the Python version that needs to be supported it might not be possible to switch. I haven't tested whether the `from __future__ import annotations` would backport the switch to `ABC`s

Reply:
Ah, very true, I've updated it to use `typing.Iterable` (which was only deprecated when `collections.abc.Iterable` was made generic in 3.9, which we're not using quite yet).